### PR TITLE
Extend custom testing framework style behaviour to function calls

### PR DESCRIPTION
**Pull request by josephw**
We have a number of tests using [`jasmine-promise-tools`](https://www.npmjs.com/package/jasmine-promise-tools)'s `willResolve` to decorate the function:

    describe("does something really long and complicated so I have to write a very long name for the describe block", willResolve(() => {
      console.log("hello!");
    }));

However, this doesn't currently get the special treatment added in #159, and ends up split:

    describe(
      "does something really long and complicated so I have to write a very long name for the describe block",
      willResolve(() => {
        console.log("hello!");
      })
    );

This PR extends the treatment so that the top-level arguments are kept on the same line:

    describe("does something really long and complicated so I have to write a very long name for the describe block", willResolve(
      () => {
        console.log("hello!");
      }
    ));

Ideally, we'd keep the arrow function on the same line to match the original formatting, but this initial change is already an improvement.

Are you open to a change here? I'm keen to figure out if I should put my effort into improving the rules for this case or learning to love the split lines.

### CODE REVIEW

File: tests/test_declarations/__snapshots__/jsfmt.spec.js.snap [hunk 1]
```diff
@@ -113,6 +117,12 @@ describe(\`does something really long and complicated so I have to write a very
   });
 });
 
+describe("does something really long and complicated so I have to write a very long name for the describe block", willResolve(
+  () => {
```
Comment (vjeux) [inline tests/test_declarations/__snapshots__/jsfmt.spec.js.snap]: Looks like it puts the arrow function on the next line anyway, so it doesn't actually solve your issue :(

**Comment by vjeux** [comment]
I'm not really sure that it's an improvement since it indents things twice. Inlining the second function is going to get trickier and I'm not sure it is worth the complexity given the very low usage of jasmine-promise-tools. Thanks for the pull request and the test though, I really appreciate!

Super biased but Jest has really nice facilities for testing async code, you can return a promise directly from it without those wrappers, you can use `.resolve` or even use async functions directly. https://facebook.github.io/jest/docs/tutorial-async.html#resolves

File: tests/test_declarations/__snapshots__/jsfmt.spec.js.snap (hunk 1 above)
Comment (josephw) [inline tests/test_declarations/__snapshots__/jsfmt.spec.js.snap]: It's not perfect. It solves keeping the string on the same line as the function name, which is the biggest visual nit, but fumbles after the comma.

**Comment by josephw** [comment]
Thanks for the feedback. I'm keen to adopt Prettier, so I'm looking at the lowest-friction ways to introduce it to an existing codebase.

> I'm not sure it is worth the complexity given the very low usage of jasmine-promise-tools
> ...
> Super biased but Jest has really nice facilities for testing async code

I'll certainly consider Jest, but changing our existing suite is going to be a slow path to victory. Given our use of Jasmine, it looks like we'll be best off excluding tests from Prettification rather than trying to address these edge cases.

Thanks again.

**Comment by vjeux** [comment]
After thinking through it more, I'm sorry but I'll have to refuse this. I'm trying to add the minimum number of edge cases for those as they are always hard to understand. It seems like the lib it is affecting is not very popular and the fix isn't even solving the main issue: the double indent.

Sorry :(

**Comment by jbunton-atlassian** [comment]
Hi @vjeux, would you be receptive to a PR which completely resolved the formatting issue?

Note that I've used this style for things like `co.wrap` to simulate async/await with generators. I think it's a generally useful pattern and it's unfortunate that prettier mangles the formatting of it so badly :(

**Comment by vjeux** [comment]
Can you show an example of how you use `co.wrap` with this pattern?

**Comment by jbunton-atlassian** [comment]
```
it('some test description', willResolve(co.wrap(function*() {
  const response = yield makeHttpRequest();
  expect(response.body).toEqual({foo: 'bar'});
})));
```

So even if using a test runner which doesn't need `willResolve` this `co.wrap` is still useful if the async/await syntax is not available.

I think being able to compose helper functions with the test function is a generally useful feature, having it formatted properly would be nice :)
